pkg/lib/bundle/validate.go

```diff
@@ -0,0 +1,142 @@
+package bundle
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/operator-framework/operator-registry/pkg/containertools"
+
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+// imageValidator is a struct implementation of the Indexer interface
+type imageValidator struct {
+	imageReader containertools.ImageReader
+	logger      *log.Entry
+}
+
+// PullBundleImage shells out to a container tool and pulls a given image tag
+// Then it unpacks the image layer filesystem contents and pushes the contents
+// to a specified directory for further validation
+func (i imageValidator) PullBundleImage(imageTag, directory string) error {
+	i.logger.Debug("Pulling and unpacking container image")
+
+	return i.imageReader.GetImageData(imageTag, directory)
+}
+
+// ValidateBundle takes a directory containing the contents of a bundle and validates
+// the format of that bundle for correctness
+func (i imageValidator) ValidateBundle(directory string) error {
+	var manifestsFound, metadataFound bool
+	var annotationsDir, manifestsDir string
+	var annotationErrors []error
+	var formatErrors []error
+
+	items, _ := ioutil.ReadDir(directory)
+	for _, item := range items {
+		if item.IsDir() {
+			switch s := item.Name(); s {
```

Comment:
Direct matching seems brittle here since the manifests or metadata dirs can be children, unless I'm missing something. `strings.HasSuffix(s, strings.TrimSuffix(ManifestsDir, "/"))` would be a better comparison.